# Define parameters to be read from 'NxtRos2Setup' node:
#   Sensors (optional):
#     - port: 1-4, must be unique
#       - sensor_type: 'touch', 'reflected_light', 'color', 'ultrasonic'
#       - sensor_name: unique string
#       - frame_id: optional, sets the message.header.frame_id to frame_id
#
#   Motors (optional):
#     - port: A, B or C, must be unique
#       - motor_type: 'other', 'wheel_motor_l', 'wheel_motor_r'
#       - motor_name: unique string
#       - motor_mimic_name: unique string
#       - motor_mmic_gear_ratio: > 0 -> mimic joint turns the same direction
#                                as the motor joint, < 0 turns the the opposite
#                                direction
#       - invert_direction: inverts the motors direction, used to configure
#                           motor orientation, check readme for more details
#
#   Robot dimensions:
#     (optional, only required if you have a two wheel differentially driven
#     robot and need the controller or odometry nodes. If you have a URDF
#     file describing the robot, the values must match for the visualization
#     to work properly.)
#     - robot_dimensions:
#       - axle_track: distance between wheel_motor_l and wheel_motor_r in meters
#       - wheel_radius: wheel radius in meter
#       - rad_per_s_to_effort: multiplier

/nxt_ros_setup:
  ros__parameters:
    devices:
      1:
        sensor_type: touch
        sensor_name: touch_sensor
      3:
        sensor_type: reflected_light
        sensor_name: reflected_light_sensor
      4:
        sensor_type: ultrasonic
        sensor_name: ultrasonic_sensor
        frame_id: radar_link
      A:
        motor_type: other
        motor_name: radar_motor
        motor_mimic_name: radar
        motor_mimic_gear_ratio: -0.33333
        invert_direction: false
      B:
        motor_type: wheel_motor_l
        motor_name: wheel_motor_l
        motor_mimic_name: wheel_l
        motor_mimic_gear_ratio: -0.33333
        invert_direction: true
      C:
        motor_type: wheel_motor_r
        motor_name: wheel_motor_r
        motor_mimic_name: wheel_r
        motor_mimic_gear_ratio: -0.33333
        invert_direction: true
    robot_dimensions:
      axle_track: 0.135
      wheel_radius: 0.022
      rad_per_s_to_effort: 5.86

# Define node specific/local parameters (has to match with motor/sensor_name above)
/ultrasonic_sensor:
  ros__parameters:
    # Default values for LEGO Mindstorms NXT 2.0 ultrasonic sensor
    field_of_view: 0.5235988 # 30 degrees
    max_range: 2.54 # meters
    min_range: 0.07 # meters

/reflected_light_sensor:
  ros__parameters:
    rgb_color:
      - 0.0
      - 0.0
      - 0.0

/joint_state_aggregator:
  ros__parameters:
    related_js_time_delta: 1000000000 # nanoseconds
